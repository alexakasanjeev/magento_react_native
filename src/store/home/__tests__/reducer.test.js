import homeReducer, { INITIAL_STATE } from '../reducer';
import { MAGENTO } from '../../../constants';
import Status from '../../../magento/Status';

describe('home reducer', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = homeReducer(undefined, {});

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle init app failure', () => {
    // Setup
    const action = {
      type: MAGENTO.INIT_APP_FAILURE,
      payload: {
        errorMessage: 'Invalid intregation token'
      }
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle home data loading', () => {
    // Setup
    const action = {
      type: MAGENTO.HOME_DATA_LOADING,
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle home data success', () => {
    // Setup
    const action = {
      type: MAGENTO.HOME_DATA_SUCCESS,
      payload: {
        30: {
          // dummy object
        },
        31: {
          // dummy object
        },
        slider: [
          // dummy object
        ]
      }
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle home data failure', () => {
    // Setup
    const action = {
      type: MAGENTO.HOME_DATA_FAILURE,
      payload: {
        errorCode: 23,
        errorMessage: 'Inavlid block id.'
      }
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle featured category products loading', () => {
    // Setup
    const categoryId = 30;
    const action = {
      type: MAGENTO.FEATURED_CATEGORY_PRODUCTS_LOADING,
      payload: {
        categoryId,
      }
    };
    const initialState = {
      ...INITIAL_STATE,
      [categoryId]: {
        errorMessage: '',
        items: [],
        status: Status.DEFAULT,
      }
    };

    // Exercise
    const newState = homeReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle featured category products success', () => {
    // Setup
    const action = {
      type: MAGENTO.FEATURED_CATEGORY_PRODUCTS_SUCCESS,
      payload: {
        categoryId: 30,
        products: {
          items: [
            {
              // dummy object
              id: 968
            }
          ]
        }
      }
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle featured category products failure', () => {
    // Setup
    const action = {
      type: MAGENTO.FEATURED_CATEGORY_PRODUCTS_FAILURE,
      payload: {
        categoryId: 30,
        errorMessage: 'Inavlid category Id.'
      }
    };

    // Exercise
    const newState = homeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle home update configurable product detail success', () => {
    // Setup
    const action = {
      type: MAGENTO.HOME_UPDATE_CONF_PRODUCT_SUCCESS,
      payload: {
        sku: 'MSH12',
        children: [{}, {}, {}],
        price: {
          starting: 27,
          ending: 27,
        }
      }
    };
    const initialState = {
      ...INITIAL_STATE,
      extra: {
        MSH07: {
          price: {
            starting: 35,
            ending: 35,
          },
          children: [{}, {}]
        }
      }
    };

    // Exercise
    const newState = homeReducer(initialState, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });
});
