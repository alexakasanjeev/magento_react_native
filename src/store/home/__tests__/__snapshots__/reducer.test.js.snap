// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home reducer should handle featured category products failure 1`] = `
Object {
  "30": Object {
    "errorMessage": "Inavlid category Id.",
    "status": "ERROR",
  },
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "NOT_STARTED",
}
`;

exports[`home reducer should handle featured category products loading 1`] = `
Object {
  "30": Object {
    "errorMessage": "",
    "items": Array [],
    "status": "LOADING",
  },
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "NOT_STARTED",
}
`;

exports[`home reducer should handle featured category products success 1`] = `
Object {
  "30": Object {
    "items": Array [
      Object {
        "id": 968,
      },
    ],
    "status": "SUCCESS",
  },
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "NOT_STARTED",
}
`;

exports[`home reducer should handle home data failure 1`] = `
Object {
  "errorMessage": "23
Inavlid block id.",
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "ERROR",
}
`;

exports[`home reducer should handle home data loading 1`] = `
Object {
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "LOADING",
}
`;

exports[`home reducer should handle home data success 1`] = `
Object {
  "30": Object {},
  "31": Object {},
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "SUCCESS",
}
`;

exports[`home reducer should handle home update configurable product detail success 1`] = `
Object {
  "extra": Object {
    "MSH07": Object {
      "children": Array [
        Object {},
        Object {},
      ],
      "price": Object {
        "ending": 35,
        "starting": 35,
      },
    },
    "MSH12": Object {
      "children": Array [
        Object {},
        Object {},
        Object {},
      ],
      "price": Object {
        "ending": 27,
        "starting": 27,
      },
    },
  },
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "NOT_STARTED",
}
`;

exports[`home reducer should handle init app failure 1`] = `
Object {
  "errorMessage": "Invalid intregation token",
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "ERROR",
}
`;

exports[`home reducer should return the initial state 1`] = `
Object {
  "extra": Object {},
  "featuredProducts": Object {},
  "slider": Array [],
  "status": "NOT_STARTED",
}
`;
